File: MCRRTImp/src/test/java/lab/mars/MCRRTImp/Vector2BasedImp/TestCircleObstacle.java
package lab.mars.MCRRTImp.Vector2BasedImp;

import lab.mars.MCRRTImp.Vector2BasedImp.CircleObstacle;
import lab.mars.MCRRTImp.Vector2BasedImp.Vector2;
import org.junit.Test;

public class TestCircleObstacle {

    @Test
    public void testContains() {
        CircleObstacle obstacle = new CircleObstacle(10, 10, 15);
        Vector2 planeCenter = new Vector2(0, 0);
        Vector2 plane2Center = new Vector2(3, 3);
        assert !obstacle.contains(planeCenter);
        assert obstacle.contains(plane2Center);
    }

}

File: MCRRTImp/src/test/kotlin/lab/mars/MCRRTImp/Vector2BasedImp/TestCircleObstacle.kt
package lab.mars.MCRRTImp.Vector2BasedImp

import infrastructure.Vector2BasedImp.CircleObstacle
import infrastructure.Vector2BasedImp.Vector2
import org.junit.Test

class TestCircleObstacle {

    @Test
    fun testContains() {
        val obstacle = CircleObstacle(10.0, 10.0, 15.0)
        val planeCenter = Vector2(0.0, 0.0)
        val plane2Center = Vector2(3.0, 3.0)
        assert(obstacle.contains(planeCenter))
        assert(obstacle.contains(plane2Center))
    }

}
